BrainPortal/app/controllers/userfiles_controller.rb

```diff
@@ -325,8 +327,11 @@ def show #:nodoc:
     @sync_status        = 'ProvNewer' # same terminology as in SyncStatus
     state               = @userfile.local_sync_status
     @sync_status        = state.status if state
-    @viewer             = @userfile.find_viewer(params[:viewer]) if params[:viewer].present?
-    @viewer           ||= @userfile.viewers.first
+    @viewer             = @userfile.find_viewer_with_applied_conditions(params[:viewer]) if params[:viewer].present?
+
+    # binding.pry
+    @viewers            = @userfile.viewers_with_applied_conditions || []
+    @viewer           ||= @viewers.detect { |v| v.errors.empty?} || @viewers.first
```

Comment:
Doesn't the `||` provide a viewer that has errors in it, if all viewers have errors?!?

Reply:
Resolved after discussion